src/ristretto.rs

```diff
@@ -1097,6 +1099,61 @@ impl Zeroize for RistrettoPoint {
     }
 }
 
+// ------------------------------------------------------------------------
+// elliptic-curve traits
+// ------------------------------------------------------------------------
+
+#[cfg(feature = "elliptic-curve")]
+impl group::Group for RistrettoPoint {
```

Comment:
For what it's worth those traits are actually from https://github.com/zkcrypto/group and re-exported by the `elliptic-curve` crate /cc @str4d

To even be able to implement `PrimeCurveArithmetic`, there are a number of other traits that need to be impl'd first, including `Group`, then `PrimeGroup`, then `PrimeCurve`. So this is the right place to start.

Reply:
For the record (and in any future PR adding impls of the `group` traits), Ristretto would implement `PrimeGroup` but not `PrimeCurve`, precisely because it is not an elliptic curve.

Reply:
Right, this is actually rather problematic because `PublicKey`/`SecretKey`, DH and hash2curve are only implemented for `PrimeCurve` ... this will need a rather big adjustment.

Reply:
> Right, this is actually rather problematic because `PublicKey`/`SecretKey`, DH and `hash2curve` are only implemented for `PrimeCurve` ... this will need a rather big adjustment.

`hash2curve` is kinda by definition only for curves, so it doesn't make sense to use with Ristretto (which instead has its own hash-to-group API).

The rest can and should be changed to take `PrimeGroup` instead of `PrimeCurve`.

Reply:
https://www.ietf.org/archive/id/draft-irtf-cfrg-hash-to-curve-16.html#name-hashing-to-ristretto255
> Since hash_to_ristretto255 is not a hash-to-curve suite, it does not have a Suite ID.

We will see how we solve this when we get there ... but yeah, we need some kind of hash2curve trait that works for Ristretto255 as specified in the spec.

Reply:
Thank you for all the input, it's very helpful.